core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -74,9 +78,12 @@ class MetadataCache(brokerId: Int) extends Logging {
   // Otherwise, return LEADER_NOT_AVAILABLE for broker unavailable and missing listener (Metadata response v5 and below).
   private def getPartitionMetadata(snapshot: MetadataSnapshot, topic: String, listenerName: ListenerName, errorUnavailableEndpoints: Boolean,
                                    errorUnavailableListeners: Boolean): Option[Iterable[MetadataResponse.PartitionMetadata]] = {
-    snapshot.partitionStates.get(topic).map { partitions =>
-      partitions.map { case (partitionId, partitionState) =>
-        val topicPartition = new TopicPartition(topic, partitionId.toInt)
+    val result = new ArrayBuffer[MetadataResponse.PartitionMetadata]
```

Comment:
The stream APIs used here are implemented in scala classes. Do we know they are also problematic?

Reply:
Leaving this as such since it makes it easier to debug code with breakpoints.